docs/about-mono/releases/5.0.0.md

```diff
@@ -25,6 +25,39 @@ Mono.Simd assembly is deprecated in this release, and it will be
 removed in +2 releases. Users should transition to using the Vector
 classes, which are part of the official .net class libraries.
 
+### Thread.Abort harderning ###
+
+The runtime was hardened to further its resilience in face of thread aborts.
+It now won't abort in-flight class constructors.
+Thread abort is important for application using multiple AppDomains.
+
+### Memory usage ###
+
+Improvements in the runtime representation of the type system should yield a 5% memory saving
+in most workloads. It was archieved by using a more efficient representation for arrays and generic instances.
+
+A linearalizable property bag was added to replace the current mechanism of rare fields in runtime metadata.
+More rarely used fields were moved to use it as they now use significantly less memory when used.
+
+### Lazy array interfaces ###
+
+One curious aspect of C# is that arrays implement invarient interfaces as if they were covariant. This happens for `IList<T>, ICollection<T> and IEnumerable<T>` which means, for example, that `string[]` implements both `IList<string>` and `IList<object>`.
+
+Mono traditionally implemented this by creating the runtime-side metadata for all of those interfaces and that came with an extraordinary memory cost of creating a lot of interfaces that never ended up being referenced by C# code.
+
+With mono 5.0 we now treat those interfaces as magic/special and use a different casting codepath. This allow them to be lazily implemented in arrays, which can save a lot of memory in LINQ-heavy workloads. As part of this work we refactored the casting code in the JIT to be simpler and more maintainable.
+
+### Symbolification ###
+
+The runtime crash stack traces now include information on managed stack traces to allow them to be symbolificated. Tooling was adjusted to handle the slightly different format.
+
+### Workaround for netstandard nugets on desktop ###
+
+Some netstandard NuGets like System.IO.Compression resolve to assemblies that won't work with mono on Desktop. To handle this we implemented a series of workarounds:
```

Comment:
For what it's worth, `System.IO.Compression` from NuGet works fine if you also install `runtime.native.System.IO.Compression` and symlink `libSystem.IO.Compression.Native.so` to `System.IO.Compression.Native.so`. Not ideal but it does work.